apps/condo/domains/common/utils/serverSchema/index.js

```diff
@@ -162,49 +180,63 @@ const loadListByChunks = async ({
     chunkSize = 100,
     limit = 100000,
     chunkProcessor = (chunk) => chunk,
+    timeoutDuration,
 }) => {
     if (chunkSize < 1 || limit < 1) throw new Error('Both chunkSize and limit should be > 0')
     if (chunkSize > 100) throw new Error('chunkSize is too large, max 100 allowed')
+
     let skip = 0
     let maxIterationsCount = Math.ceil(limit / chunkSize)
     let newChunk = []
     let all = []
     let newChunkLength
     let haveWarnedAboutTooManyObjs = false
 
-    do {
-        newChunk = await list.getAll(context, where, { sortBy, first: chunkSize, skip: skip })
-        newChunkLength = newChunk.length
+    let timeoutFlag = { timeoutOccurred: false }
 
-        if (newChunkLength > 0) {
-            if (isFunction(chunkProcessor)) {
-                newChunk = chunkProcessor.constructor.name === 'AsyncFunction'
-                    ? await chunkProcessor(newChunk)
-                    : chunkProcessor(newChunk)
-            }
+    const processChunks = async () => {
+        do {
+            if (timeoutFlag.timeoutOccurred) break
 
-            skip += newChunkLength
-            all = all.concat(newChunk)
-        }
+            newChunk = await list.getAll(context, where, { sortBy, first: chunkSize, skip: skip })
+            newChunkLength = newChunk.length
 
-        if ((!haveWarnedAboutTooManyObjs) && (all && Array.isArray(all) && all.length > 1000)) {
-            logger.warn({
-                msg: 'tooManyReturned',
-                functionName: 'loadListByChunks',
-                schemaName: list.key,
-                data: {
-                    limit: 1000,
-                    loadListByChunksArgs: { where },
-                },
-            })
-            haveWarnedAboutTooManyObjs = true
-        }
+            if (newChunkLength > 0) {
+                if (isFunction(chunkProcessor)) {
+                    newChunk = chunkProcessor.constructor.name === 'AsyncFunction'
+                        ? await chunkProcessor(newChunk)
+                        : chunkProcessor(newChunk)
+                }
+
+                skip += newChunkLength
+                all = all.concat(newChunk)
+            }
 
-    } while (--maxIterationsCount > 0 && newChunkLength)
+            if ((!haveWarnedAboutTooManyObjs) && (all && Array.isArray(all) && all.length > 1000)) {
+                logger.warn({
+                    msg: 'tooManyReturned',
+                    functionName: 'loadListByChunks',
+                    schemaName: list.key,
+                    data: {
+                        limit: 1000,
+                        loadListByChunksArgs: { where },
+                    },
+                })
+                haveWarnedAboutTooManyObjs = true
+            }
+
+        } while (--maxIterationsCount > 0 && newChunkLength && !timeoutFlag.timeoutOccurred)
+    }
+    
+    await Promise.race([
+        processChunks(),
+        createTimeoutPromise(timeoutDuration),
+    ])
```

Comment:
I'm not sure, but can you check if Promise.race method discards the result of the second promise? 
I have been worrying about possible memory leaks and memory cleaning here